ql/ql/src/codeql/GlobalValueNumbering.qll

```diff
@@ -5,7 +5,7 @@ private import codeql_ql.ast.internal.Builtins
 
 private newtype TValueNumber =
   TVariableValueNumber(VarDecl var) { variableAccessValueNumber(_, var) } or
-  TFieldValueNumber(VarDecl var) { fieldAccessValueNumber(_, var) } or
+  TFieldValueNumber(FieldDecl var) { fieldAccessValueNumber(_, var) } or
```

Comment:
is this an unrelated drive-by bugfix?

Reply:
I think `var intanceof FieldDecl` was implied by `fieldAccessValueNumber` already. So this is just making the branch more readable, I think.

Reply:
Yes, I was trying to make it clear when something is a `FieldDecl` and when something is a `VarDecl`.  

There is a lot of drive-by "Make `FieldDecl` and it's uses better"-fixes in this PR.